components/push/src/communications.rs

```diff
@@ -256,39 +258,60 @@ impl Connection for ConnectHttp {
     }
 
     /// Drop a channel and stop receiving updates.
-    ///
-    /// A `None` channel_id indicates that we are unsubscribing
-    /// from all channels
-    fn unsubscribe(&mut self, channel_id: Option<&str>) -> error::Result<bool> {
+    fn unsubscribe(&self, channel_id: &str) -> error::Result<bool> {
         if self.auth.is_none() {
             return Err(CommunicationError("Connection is unauthorized".into()).into());
         }
         if self.uaid.is_none() {
             return Err(CommunicationError("No UAID set".into()).into());
         }
         let options = self.options.clone();
-        let mut url = format!(
-            "{}://{}/v1/{}/{}/registration/{}",
+        let url = format!(
+            "{}://{}/v1/{}/{}/registration/{}/subscription/{}",
             &options.http_protocol.unwrap(),
             &options.server_host,
             &options.bridge_type.unwrap(),
             &options.sender_id,
             &self.uaid.clone().unwrap(),
+            channel_id,
         );
-        if let Some(channel_id) = channel_id {
-            url = format!("{}/subscription/{}", url, channel_id)
-        }
         if &self.options.sender_id == "test" {
             return Ok(true);
         }
         let response = Request::delete(Url::parse(&url)?)
             .headers(self.headers()?)
             .send()?;
         self.check_response_error(&response)?;
-        if channel_id.is_none() {
-            self.uaid = None;
-            self.auth = None;
+        Ok(true)
+    }
+
+    /// Drops all channels and stops receiving notifications.
+    /// this also wipes the `uaid` and the `auth` fields.
+    fn unsubscribe_all(&mut self) -> error::Result<bool> {
+        if self.auth.is_none() {
+            return Err(CommunicationError("Connection is unauthorized".into()).into());
+        }
+        if self.uaid.is_none() {
+            return Err(CommunicationError("No UAID set".into()).into());
         }
+        let options = self.options.clone();
+        let url = format!(
+            "{}://{}/v1/{}/{}/registration/{}",
+            &options.http_protocol.unwrap(),
+            &options.server_host,
+            &options.bridge_type.unwrap(),
+            &options.sender_id,
+            &self.uaid.clone().unwrap(),
+        );
```

Comment:
There's a good deal of duplicated code between this and `unsubscribe()`, maybe it could be moved to a private helper function.

Reply:
Moves the formating bits out, I'm keeping the error checking at the top, since that will be refactored with the rest of #4372 😄